docs/development/zod.md

```diff
@@ -0,0 +1,356 @@
+### Table of Contents
+
+- [Zod schema guideline](#zod-schema-guideline)
+  - [When and where to use Zod](#when-and-where-to-use-zod)
+  - [Technical guide](#technical-guide)
+    - [Use `schema.ts` files for Zod schemas](#use-schemats-files-for-zod-schemas)
+    - [Name schemas without any `Schema` suffix](#name-schemas-without-any-schema-suffix)
+    - [Inferred types](#inferred-types)
+    - [Specify only necessary fields](#specify-only-necessary-fields)
+    - [Use `Json`, `Yaml` and `Toml` for string parsing](#use-json-yaml-and-toml-for-string-parsing)
+    - [Use `.transform()` method to process parsed data](#use-transform-method-to-process-validated-data)
+      - [Rename and move fields at the top level transform](#rename-and-move-fields-at-the-top-level-transform)
+    - [Stick to permissive behavior when possible](#stick-to-permissive-behavior-when-possible)
+      - [Use `.catch()` to force default values](#use-catch-to-force-default-values)
+      - [Use `LooseArray` and `LooseRecord` to filter out incorrect values from collections](#use-loosearray-and-looserecord-to-filter-out-incorrect-values-from-collections)
+    - [Combining with `Result` class](#combining-with-result-class)
+    - [Combining with `Http` class](#combining-with-http-class)
```

Comment:
Is this ToC really needed? I think it risks getting out of date, makes it harder to refactor/update, etc. I would only like it if it can be automated

Reply:
Let's remove it when we notice it's outdated the second or third time? Links are checked by CI, and mantaining it should not that hard compared to spending time to automate it.

IMO yes, TOC is useful, because not too much people may know GitHub has automatic TOC dropdown in the Markdown files.